src/historywork/GetRemoteFileWork.cpp

```diff
@@ -37,13 +41,23 @@ GetRemoteFileWork::getCommand()
     releaseAssert(mCurrentArchive);
     releaseAssert(mCurrentArchive->hasGetCmd());
     auto cmdLine = mCurrentArchive->getFileCmd(mRemote, mLocal);
-
+    // getCommand is called from RunCommandWork::onRun immediately before
+    // running the command.  Record the time here so that we can calculate
+    // throughput in onSuccess.
+    mDownloadStart = std::chrono::steady_clock::now();
+    if (getRetryCount() > 0)
+    {
+        mRetryRemoteFile.inc();
+    }
+    CLOG_INFO(History, "Downloading file '{}' from archive '{}', COMMAND='{}'",
+              mRemote, mCurrentArchive->getName(), cmdLine);
     return CommandInfo{cmdLine, std::string()};
 }
 
 void
 GetRemoteFileWork::onReset()
 {
+    mDownloadStart = std::chrono::steady_clock::now();
```

Comment:
I don't think this is the correct place to set the start timer. I think it would be cleaner to wrap `mDownloadStart` in an optional and clear it on reset. Then in the path that begins the download, you can check if `mDownloadStart` is empty and start the timer then. This should be cleaner also if you move this up to `GetHistoryArchiveStateWork`